### PR TITLE
Minimal testing framework

### PR DESCRIPTION
**Pull request by Luthaf**
WIP toward #21 

Minimal basis of testing framework using [chai](https://www.chaijs.com/) (assertions), [mocha](https://mochajs.org/) (unit test framework), and [jsdom](https://github.com/jsdom/jsdom) (mock implementation of the DOM api running on node). 

It looks like we could be able to do quite a bit just with this, since jsdom is giving us "full" DOM access. The only thing that would still benefit from cypress or something similar is changes happening across panels (click to set the new active, etc.)

### CODE REVIEW

**Comment by Luthaf** [comment]
This should be ready for review. Once merged, other can chime in to implement tests for existing code, and we can start requiring tests for new code!

File: src/map/options.ts
Comment (rosecers) [inline src/map/options.ts:76]: What is the purpose of this file being included in this PR? These changes look like cleanups.

File: tests/jsdom.ts
Comment (rosecers) [inline tests/jsdom.ts:4]: Can you add some documentation here to point to what this test does?

File: tests/map/options.test.ts
Comment (rosecers) [inline tests/map/options.test.ts:29]: I'm not so familiar with this status -- can you explain what the successive `it(...)` statements do?

**Review by rosecers — COMMENTED** [review]
I don't see anything immediate to change -- most of my comments are questions or general comments.

File: src/map/options.ts
Comment (Luthaf) [reply src/map/options.ts:76]: This is a fix for one of the tests I added, which was failling, namely checking that calling `remove` doe indeed remove everything.

File: tests/map/options.test.ts
Comment (Luthaf) [reply tests/map/options.test.ts:29]: So the way mocha works is that test cases are introduced as calls to the `it` function, with a first argument describing the test case, and the second argument executing the test itself. If there is no second argument, as here, the test is skipped and considered "to be written later". 

I added these mostly as a TODO list for myself, but I can also keep them out of the code for now.

Comment (rosecers) [reply tests/map/options.test.ts:29]: It might be best to either omit in this PR or write the coinciding tests

File: src/map/options.ts
Comment (rosecers) [reply src/map/options.ts:76]: Does it belong in this PR or should it be split off?

**Comment by rosecers** [comment]
Is this active?

**Comment by Luthaf** [comment]
Working on it right now =)

File: tests/map/options.test.ts
Comment (Luthaf) [reply tests/map/options.test.ts]: Removed, I'll send another PR implementing these tests

File: tests/jsdom.ts
Comment (Luthaf) [reply tests/jsdom.ts:7]: done!

File: src/map/options.ts
Comment (Luthaf) [reply src/map/options.ts:76]: I this it belongs to here, tests and corresponding fix are in the second commit

**Comment by Luthaf** [comment]
This should be good to go!

**Comment by Luthaf** [comment]
I'll merge this on Monday if nobody disagree!

**Review by rosecers — APPROVED** [review]

**Pull request merged**
